### PR TITLE
[v6r19] Saving extension in pilots

### PR DESCRIPTION
**Pull request by fstagni**
BEGINRELEASENOTES

*WMS
FIX: pilot commands will add /DIRAC/Extensions=extensions if requested

*Workflow
FIX: application log name can also come from step_commons.get['logFile']

ENDRELEASENOTES

and some pylints as usual

### CODE REVIEW

File: Workflow/Modules/ModuleBase.py [hunk 1]
```diff
@@ -303,22 +303,14 @@ def _resolveInputStep( self ):
...
 
-    if self.step_commons.has_key( 'applicationVersion' ) and self.step_commons['applicationVersion']:
-      self.applicationVersion = self.step_commons['applicationVersion']
-    else:
-      self.applicationVersion = 'Unknown'
+    self.applicationVersion = self.step_commons.get('applicationVersion', 'Unknown')
 
-    self.applicationLog = self.step_commons.get('applicationLog', self.applicationLog)
+    self.applicationLog = self.step_commons.get('applicationLog', 
+                                                self.step_commons.get('logFile', self.applicationLog))
```
Comment (chaen) [inline Workflow/Modules/ModuleBase.py]: Why also `logFile` ?

File: Workflow/Modules/Script.py [hunk 2]
```diff
@@ -68,7 +68,8 @@ def _setCommand( self ):
     if os.path.exists( os.path.basename( self.executable ) ):
       self.executable = os.path.basename( self.executable )
       if not os.access( '%s/%s' % ( os.getcwd(), self.executable ), 5 ):
-        os.chmod( '%s/%s' % ( os.getcwd(), self.executable ), stat.S_IRWXU | stat.S_IRGRP | stat.S_IXGRP | stat.S_IROTH | stat.S_IXOTH )
+        os.chmod( '%s/%s' % ( os.getcwd(), self.executable ),
```
Comment (chaen) [inline Workflow/Modules/Script.py]: Mind adding a comment with the octal form of all these flags ? :D

Comment (fstagni) [reply Workflow/Modules/Script.py]: Linking to https://docs.python.org/2/library/stat.html#stat.S_IRWXU is enough?

File: AccountingSystem/Client/Types/BaseAccountingType.py [hunk 3]
```diff
@@ -42,7 +38,7 @@ def checkType( self ):
     self.fieldsList.extend( self.keyFieldsList )
     self.fieldsList.extend( self.valueFieldsList )
     if len( self.valuesList ) != len( self.fieldsList ):
-      self.valuesList = [ None for i in self.fieldsList ]
+      self.valuesList = [ None for _i in self.fieldsList ]
```
Comment (chaen) [inline AccountingSystem/Client/Types/BaseAccountingType.py]: Really minor, but I suppose that ``` [None] * len(self.fieldsList) ``` is faster and more memory efficient. But we do not really care for 4 elements I suppose :)

File: AccountingSystem/private/DBUtils.py [hunk 4]
```diff
@@ -31,11 +30,13 @@ def _retrieveBucketedData( self,
                        ( "%s, %s", ( "field1name", "field2name", "field3name" )
     """
     validCondDict = {}
-    if type( condDict ) == types.DictType:
+    if isinstance(condDict, dict):
       for key in condDict:
-        if type( condDict[ key ] ) in ( types.ListType, types.TupleType ) and len( condDict[ key ] ) > 0:
+        if isinstance(condDict[ key ], (list, tuple)) and len(condDict[key]):
```
Comment (chaen) [inline AccountingSystem/private/DBUtils.py]: Can ```len(condDict[key])``` be replaced by just ```condDict[key]``` ?

File: Workflow/Modules/Script.py (hunk 2 above)
Comment (chaen) [reply Workflow/Modules/Script.py]: well, sure, we can understand from it, but most human understand better ```640``` than the flag sequence

File: AccountingSystem/private/DBUtils.py (hunk 4 above)
Comment (fstagni) [reply AccountingSystem/private/DBUtils.py]: Probably yes, for this specific case. I will change it.

**Comment by chaen** [comment]
review OK

File: Workflow/Modules/ModuleBase.py [hunk 5]
```diff
@@ -303,22 +303,14 @@ def _resolveInputStep( self ):
 
     self.stepName = self.step_commons['STEP_INSTANCE_NAME']
 
-    if 'executable' in self.step_commons and self.step_commons['executable']:
-      self.executable = self.step_commons['executable']
-    else:
-      self.executable = 'Unknown'
+    self.executable = self.step_commons.get('executable', 'Unknown')
```
Comment (phicharp) [inline Workflow/Modules/ModuleBase.py]: This is not strictly the asme thing, in case `step_commons['executable']` is `None` for example

**Comment by phicharp** [comment]
Still spaces after [ or ( and before ] or )

File: AccountingSystem/Client/Types/BaseAccountingType.py [hunk 6]
```diff
@@ -1,15 +1,14 @@
...
-import types
 from DIRAC import S_OK, S_ERROR
 from DIRAC.Core.Utilities import Time
 from DIRAC.Core.DISET.RPCClient import RPCClient
 from DIRAC.AccountingSystem.Client.DataStoreClient import gDataStoreClient
 
-class BaseAccountingType:
-
-  __validDataValues = ( types.IntType, types.LongType, types.FloatType, types.LongType )
+class BaseAccountingType(object):
```
Comment (atsareg) [inline AccountingSystem/Client/Types/BaseAccountingType.py]: class BaseAccountingType( object ):

File: Core/DISET/private/FileHelper.py [hunk 7]
```diff
@@ -1,23 +1,18 @@
...
 __RCSID__ = "$Id$"
 
 import os
-try:
-  import hashlib
-  md5 = hashlib
-except:
-  import md5
-import types
+import hashlib as md5
```
Comment (atsareg) [inline Core/DISET/private/FileHelper.py]: I would just do "import hashlib" and change md5->hashlib later in the code (4 cases) to definitely get rid of this historical indirection

File: AccountingSystem/Client/Types/BaseAccountingType.py (hunk 6 above)
Comment (chaen) [reply AccountingSystem/Client/Types/BaseAccountingType.py]: No, this is correct according to pep8: 
https://www.python.org/dev/peps/pep-0008/#id26

File: Core/DISET/private/FileHelper.py (hunk 7 above)
Comment (chaen) [reply Core/DISET/private/FileHelper.py]: yes !! :)

File: Workflow/Modules/ModuleBase.py (hunk 5 above)
Comment (fstagni) [reply Workflow/Modules/ModuleBase.py]: By construction this is either a string (not empty) or it doesn't exist.

File: Core/DISET/private/FileHelper.py (hunk 7 above)
Comment (fstagni) [reply Core/DISET/private/FileHelper.py]: Done

File: Core/DISET/private/Transports/SSL/SocketInfoFactory.py [hunk 8]
```diff
@@ -6,16 +6,18 @@
...
 from DIRAC.Core.Utilities.ReturnValues import S_ERROR, S_OK
 from DIRAC.Core.Utilities import Network
 from DIRAC.Core.DISET.private.Transports.SSL.SocketInfo import SocketInfo
 from DIRAC.Core.DISET.private.Transports.SSL.SessionManager import gSessionManager
 from DIRAC.FrameworkSystem.Client.Logger import gLogger
 
-if GSI.__version__ < "0.5.0":
-  raise Exception( "Required GSI version >= 0.5.0" )
+if GSI.__version__ < "0.6.5":
+  raise Exception( "Required GSI version >= 0.6.5" )
```
Comment (fstagni) [inline Core/DISET/private/Transports/SSL/SocketInfoFactory.py]: Do you see anything against this?

Comment (chaen) [reply Core/DISET/private/Transports/SSL/SocketInfoFactory.py]: I'd say no problem

**Pull request merged**